ui/src/main/java/com/hyperwallet/android/ui/transfermethod/SelectTransferMethodPresenter.java

```diff
@@ -91,75 +80,135 @@ public void loadCurrency(final boolean forceUpdate, @NonNull final String countr
             mTransferMethodConfigurationRepository.refreshKeys();
         }
 
-        mTransferMethodConfigurationRepository.getKeys(new TransferMethodConfigurationRepository.LoadKeysCallback() {
+        final TransferMethodConfigurationRepository.LoadKeysCallback loadCurrencyCallback = buildLoadCurrencyCallback(
+                countryCode);
+
+        if (mUser == null) {
+            final UserRepository.LoadUserCallback loadUserCallback = buildLoadUserCallback(loadCurrencyCallback);
+            mUserRepository.loadUser(loadUserCallback);
+        } else {
+            mTransferMethodConfigurationRepository.getKeys(loadCurrencyCallback);
+        }
+    }
+
+    @Override
+    public void loadTransferMethodTypes(final boolean forceUpdate,
+            @NonNull final String countryCode, @NonNull final String currencyCode) {
+        if (forceUpdate) {
+            mTransferMethodConfigurationRepository.refreshKeys();
+        }
+
+        final TransferMethodConfigurationRepository.LoadKeysCallback loadMethodTypesKeysCallback =
+                buildMethodKeysCallback(countryCode, currencyCode, false);
+
+        if (mUser == null) {
+            final UserRepository.LoadUserCallback loadUserCallback = buildLoadUserCallback(loadMethodTypesKeysCallback);
+            mUserRepository.loadUser(loadUserCallback);
+        } else {
+            mTransferMethodConfigurationRepository.getKeys(loadMethodTypesKeysCallback);
+        }
+    }
+
+    @Override
+    public void openAddTransferMethod(@NonNull final String country, @NonNull final String currency,
```

Comment:
This method should take one extra parameter, the profile type and in AddTransferMethodActivity add a new extra,
EXTRA_TRANSFER_METHOD_PROFILE_TYPE which will be used in AddTransferMethodPresenter#loadTransferMethodConfigurationFields. See https://confluence.site1.hyperwallet.local/display/PF/Technical+Design+-+Add+Transfer+Method
My apologies, I forgot to link the design to the ticket

Reply:
@fmattos-hw Done (see commit a8ace3d07496aef5157c990a74e76b10a157335a)